### PR TITLE
Migrate to async/await

### LINKED ISSUE #70: Move to std::future and async/await
Since `async` is now in the beta channel, and our dependencies (`tokio`, `reqwest`) have been updated to use them, we should look to migrate to using it.

I can work on this if it sounds good.

### PR DESCRIPTION
**Pull request by goweiwen**
Resolves #70

### CODE REVIEW

File: src/api/mod.rs [hunk 1]
```diff
@@ -130,187 +128,185 @@ fn generate_random_bytes(size: usize) -> String {
...
+                .await
+                .map_err(|_| backoff::Error::Transient("HTTP error"))
         }
-
-        edit_request(request_builder).build()
-            .map_err(|_| "Failed to build request")
     };
 
-    Either::B(retry_forever(client, request_builder))
+    let mut backoff = ExponentialBackoff::default();
```
Comment (angelsl) [inline src/api/mod.rs]: Could we use `backoff::Zero` instead?

Comment (indocomsoft) [reply src/api/mod.rs]: Agreed. Reason here being that Luminus would randomly return 400 to a perfectly good request for no apparent reason, sometimes with about 50% distribution of 200 and 400. Using exponential backoff results in a really annoying experience

Comment (goweiwen) [reply src/api/mod.rs]: Got it. If that's the case, I'll remove the dependency and implement the logic directly.

**Comment by indocomsoft** [comment]
Thanks for this!

**Pull request merged**